### PR TITLE
Docs Build Improvements

### PR DESCRIPTION
**Pull request by collindutter**
Refactor the documentation building process to support `dev` and `main` branches introduced by Git Flow. No longer use ReadTheDocs build on pull request functionality, we trigger it ourselves.

### CODE REVIEW

File: .github/workflows/pull-request-links-dev.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Read the Docs Preview	Dev
```
Comment (vasinov) [inline .github/workflows/pull-request-links-dev.yml:1]: Why the tab character?

File: .github/workflows/pull-request-links-prod.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: Read the Docs Preview	Prod
```
Comment (vasinov) [inline .github/workflows/pull-request-links-prod.yml:1]: Same here.

**Review by vasinov — APPROVED** [review]

File: .github/workflows/pull-request-links-dev.yml (hunk 1 above)
Comment (collindutter) [reply .github/workflows/pull-request-links-dev.yml:1]: Not sure, it's not showing like that in my editor...Let me try clearing and repushing.

File: .github/workflows/pull-request-links-prod.yml (hunk 2 above)
Comment (collindutter) [reply .github/workflows/pull-request-links-prod.yml]: Fixed

File: .github/workflows/pull-request-links-dev.yml (hunk 1 above)
Comment (collindutter) [reply .github/workflows/pull-request-links-dev.yml]: Fixed

**Review by vasinov — APPROVED** [review]

File: .github/workflows/pull-request-links-dev.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: Read the Docs Preview Prod
```
Comment (vasinov) [inline .github/workflows/pull-request-links-dev.yml:1]: Should say "Dev" here.

**Review by vasinov — CHANGES_REQUESTED** [review]

**Review by andrewfrench — APPROVED** [review]

**Review by andrewfrench — APPROVED** [review]

**Review by vasinov — APPROVED** [review]

**Pull request merged**